### ISSUE TITLE
`RET504` invalid autofix

### ISSUE THREAD

**Issue by tylerlaprade** [astral-sh/ruff#5909]
```
def foo(data):
    with contextlib.suppress(UnicodeDecodeError):
        data = data.decode()
    return data
```
is autofixed to
```
def foo(data):
    with contextlib.suppress(UnicodeDecodeError):
        return data.decode()
```
These are not equivalent. The first one will always return `data`, either decoded or the original. The second will return the decoded data if there's no exception, otherwise, it will return `None`. This introduced a bug into my code that I only caught from some miscellaneous Pyright type-checking far away from the autofix.

**Comment by zanieb** [comment]
Thanks for the report @tylerlaprade. It looks like perhaps we should not raise RET504 inside context managers?

**Comment by zanieb** [comment]
@dhruvmanila the problem is that _any_ context manager could suppress exceptions, we can't really know. I agree that `data` is defined before the context is important as well though.

**Comment by tylerlaprade** [comment]
Sourcery has been autofixing all my `try-catch-pass` blocks to the more concise `contextlib.suppress` (which I'm happy about!), but it opens up potential opportunities for this bug.

**Comment by charliermarsh** [comment]
Shameless plug that Ruff can do it too :joy: https://beta.ruff.rs/docs/rules/suppressible-exception/

**Comment by tylerlaprade** [comment]
Ah, I do have `SIM` enabled already, so I guess I have two sources of the autofixes!

**Comment by tylerlaprade** [comment]
This appears to be fixed in `0.1.15` so I will close this, thanks!
